rfcs/0014-dedicated-serving-cluster.md

```diff
@@ -0,0 +1,76 @@
+---
+feature: dedicated-serving-cluster
+authors:
+  - "Eric Fu"
+start_date: "2022/11/07"
+---
+
+# Dedicated Batch Serving Cluster
+
+## Summary
+
+Run a dedicated cluster to serve batch queries.
+
+## Motivation
+
+- **Different traffic pattern**. Batch queries, especially the high-concurrent point queries (aka. “online serving”), requires millisecond-level low latency and high availability, while streaming jobs focus more on high throughput and cost efficiency.
+- **Different access pattern on storage** of batch & streaming. Batch queries are issued by services on user side, while streaming operators always handle the changes from upstreaming system. It's possible that they have quite different distribution of access aka. hotspot. 
```

Comment:
Another access pattern is range query.

Reply:
I agree there will be more scans in batch but I don't think they are all ordered by PK. 

On the other hand, distributing data in another way rather than vnode will be much more complicated, which means it cannot leverage the existing block-level caching. If you are thinking about a KV storage without vnode, why not just introducing a deployment of TiKV or Redis? I mean, otherwise you actually have to reinvent an TiKV/Redis to do it.

Reply:
Range partition not only orders keys, but also helps pruning unnecessary scan, which is critical in our performance. I admit that things would be more complicated if we have new partitioning schema, but we can't rush code every time when customer requirements appears, our architecture should be general enough.